src/handlers/redeem_and_unstake/RedeemIntentConfirmedHandler.ts

```diff
@@ -0,0 +1,83 @@
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+import BigNumber from 'bignumber.js';
+
+import Logger from '../../Logger';
+import Message from '../../models/Message';
+import {
+  MessageDirection, MessageRepository, MessageStatus, MessageType,
+} from '../../repositories/MessageRepository';
+import ContractEntityHandler from '../ContractEntityHandler';
+import Utils from '../../Utils';
+
+/**
+ * This class handles RedeemIntentConfirmed event.
+ */
+export default class RedeemIntentConfirmedHandler extends ContractEntityHandler<Message> {
+  private messageRepository: MessageRepository;
+
+  /**
+   * @param messageRepository Instance of MessageRepository.
+   */
+  public constructor(messageRepository: MessageRepository) {
+    super();
+    this.messageRepository = messageRepository;
+  }
+
+  /**
+   * This method parse confirm redeem intent transaction and returns Message model object.
+   *
+   * @param transactions Transaction objects.
+   * @return Array of instances of Message objects.
+   */
+  public async persist(transactions: any[]): Promise<Message[]> {
+    let message: Message | null;
+    Logger.debug('Persisting Stake intent confirm records');
+    const models: Message[] = await Promise.all(transactions.map(
+      async (transaction): Promise<Message> => {
+        const messageHash = transaction._messageHash;
+        message = await this.messageRepository.get(messageHash);
+        if (message === null) {
+          message = new Message(transaction._messageHash);
+          message.gatewayAddress = transaction.contractAddress;
+          message.sender = Utils.toChecksumAddress(transaction._redeemer);
+          message.nonce = new BigNumber(transaction._redeemerNonce);
+          message.type = MessageType.Redeem;
+          message.targetStatus = MessageStatus.Undeclared;
+          message.direction = MessageDirection.AuxiliaryToOrigin;
+          message.hashLock = transaction._hashLock;
+          Logger.debug(`Creating a new message for message hash ${transaction._messageHash}`);
+        }
+        if (message.targetStatus === undefined
+          || message.targetStatus === MessageStatus.Undeclared) {
+          message.targetStatus = MessageStatus.Declared;
+        }
+        return message;
+      },
+    ));
+
+    const savePromises = [];
+    for (let i = 0; i < models.length; i += 1) {
+      Logger.debug(`Changing target status to declared for message hash ${models[i].messageHash}`);
+      savePromises.push(this.messageRepository.save(models[i]));
```

Comment:
similar to previous PR feedback, may be .catch on individual queries

Reply:
Done!